File: ios/PersonalSite/Features/Home/HomeView.swift
import SwiftUI

/// 首页：扩展版身份头 + 信号场弹幕 + Bio 打字机。
struct HomeView: View {
    /// Loader 滑出完成后开播 Bio 首访序列。
    var loaderFinished: Bool
    /// 会话内首访序列只播一次（App 级状态）。
    var bioPlayed: Bool
    var onBioPlayed: () -> Void
    var careerTimelinePlayed: Bool
    var onCareerTimelinePlayed: () -> Void
    var onShowAbout: () -> Void

    var body: some View {
        ScrollView {
            VStack(alignment: .leading, spacing: 0) {
                HomeIdentityHeader(
                    onShowAbout: onShowAbout,
                    careerTimelinePlayed: careerTimelinePlayed,
                    onCareerTimelinePlayed: onCareerTimelinePlayed
                )

                HomeBodyView(
                    loaderFinished: loaderFinished,
                    bioPlayed: bioPlayed,
                    onBioPlayed: onBioPlayed
                )
            }
            .padding(.horizontal, 32)
        }
        .scrollContentBackground(.hidden)
        .background(Color.psSurface)
    }
}

/// 扩展版身份头（仅首页）：大头像、「陈远」粗体 + 灰 @defulat-coder 两行、
/// 链接行 GitHub｜语雀 折行 ｜关于我、右上月亮，对齐 components/site-profile.tsx。
struct HomeIdentityHeader: View {
    var onShowAbout: () -> Void
    var careerTimelinePlayed: Bool
    var onCareerTimelinePlayed: () -> Void

    var body: some View {
        ProfileHeaderContent(
            onShowAbout: onShowAbout,
            careerTimelinePlayed: careerTimelinePlayed,
            onCareerTimelinePlayed: onCareerTimelinePlayed
        )
    }
}

struct HomeBodyView: View {
    var loaderFinished: Bool
    var bioPlayed: Bool
    var onBioPlayed: () -> Void

    var body: some View {
        VStack(alignment: .leading, spacing: 0) {
            SignalFieldView()
                .padding(.top, 26.4)

            BioView(
                startSignal: loaderFinished,
                shouldPlaySequence: !bioPlayed,
                onSequenceCompleted: onBioPlayed
            )
            .padding(.top, 27.2)
            .padding(.bottom, 28.8)
        }
    }
}

File: android/app/src/main/java/com/personalsite/features/home/HomeView.kt
package com.personalsite.features.home

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.personalsite.core.PSColors
import com.personalsite.core.PSTheme
import com.personalsite.features.support.ProfileHeaderContent

/** 首页：扩展版身份头 + 信号场弹幕 + Bio 打字机（对齐 iOS HomeView）。 */
@Composable
fun HomeView(
    theme: PSTheme,
    onThemeChange: (PSTheme) -> Unit,
    loaderFinished: Boolean,
    bioPlayed: Boolean,
    onBioPlayed: () -> Unit,
    careerTimelinePlayed: Boolean,
    onCareerTimelinePlayed: () -> Unit,
    onShowAbout: () -> Unit,
    reduceMotion: Boolean,
    sceneIsActive: Boolean,
) {
    Column(
        Modifier
            .verticalScroll(rememberScrollState())
            .padding(horizontal = 32.dp),
    ) {
        ProfileHeaderContent(
            theme = theme,
            onThemeChange = onThemeChange,
            onShowAbout = onShowAbout,
            careerTimelinePlayed = careerTimelinePlayed,
            reduceMotion = reduceMotion,
            onCareerTimelinePlayed = onCareerTimelinePlayed,
        )
        SignalFieldView(
            reduceMotion = reduceMotion,
            sceneIsActive = sceneIsActive,
            modifier = Modifier.padding(top = 26.4.dp),
        )
        BioView(
            startSignal = loaderFinished,
            shouldPlaySequence = !bioPlayed,
            reduceMotion = reduceMotion,
            onSequenceCompleted = onBioPlayed,
        )
        // 底部留白（对齐 iOS padding bottom 28.8）
        androidx.compose.foundation.layout.Spacer(Modifier.padding(bottom = 28.8.dp))
    }
}
